In [2]:
import numpy as np
import pandas as pd
from google.colab import files

# Question 1

In [3]:
files.upload()

Saving problem_set_1_data.csv to problem_set_1_data.csv


{'problem_set_1_data.csv': b'\xef\xbb\xbfquizzes,attendedClass,finalgrade,final_EXAM,student\r\n78.5,1,99.38,46,Linnie\xc2\xa0Lietz\r\n64.25,0,95.78,45,Mitch\xc2\xa0Mustain\r\n76.75,0,97.42,41,Salina\xc2\xa0Chavera\r\n69,1,92.5,38,Kimberely\xc2\xa0Conwell\r\n64,0,92.5,39,Zack\xc2\xa0Burk\r\n63,1,89.38,36,Almeda\xc2\xa0Rizer\r\n73.5,0,95.94,43,Damien\xc2\xa0Ellett\r\n70.5,0,95,40,Lora\xc2\xa0Sobieski\r\n78,0,96.36,42,Kieth\xc2\xa0Treanor\r\n61.25,0,88.28,37,Laverne\xc2\xa0Krok\r\n69,0,92.73,45,Ernestine\xc2\xa0Juan\r\n59,0,78.75,33,Jackelyn\xc2\xa0Baro\r\n65,0,94.84,41,Tiny\xc2\xa0Recore\r\n61.5,0,85.63,37,Ngan\xc2\xa0Hiler\r\n73,0,93.13,43,Albertha\xc2\xa0Ouk\r\n71,1,90.63,40,Mercy\xc2\xa0Dugal\r\n77,1,101.25,46,Augustine\xc2\xa0Clair\r\n75.5,0,90.94,41,Krystyna\xc2\xa0Barrentine\r\n65.5,0,92,38,Jamie\xc2\xa0Mcdowell\r\n79.5,0,96.06,37,Karole\xc2\xa0Mcgraw\r\n62.5,0,85.94,34,Dorla\xc2\xa0Canaday\r\n76.5,0,96.36,44,Fidela\xc2\xa0Leonetti\r\n57.5,0,85.31,40,Orval\xc2\xa0Ortis\r\n64,0,85,

In [4]:
data = pd.read_csv('problem_set_1_data.csv')

In [5]:
#Rename the columns to be “quiz_total”, “attended”, “final_grade”, “final_exam”, and “name”
data = data.rename(columns ={"quizzes":"quiz_total","attendedClass":'attended','finalgrade':"final_grade",'final_EXAM':'final_exam',"student":"name"})


In [6]:
#Print a “head” and “tail” of the dataframe and a summary of the dataframe. 
print(data.head())

   quiz_total  attended  final_grade  final_exam               name
0       78.50         1        99.38          46       Linnie Lietz
1       64.25         0        95.78          45      Mitch Mustain
2       76.75         0        97.42          41     Salina Chavera
3       69.00         1        92.50          38  Kimberely Conwell
4       64.00         0        92.50          39          Zack Burk


In [7]:
# print a tail of the dataframe
print(data.tail())

     quiz_total  attended  final_grade  final_exam                 name
166        73.5         0        96.25          45           Dana Malta
167        75.5         0        98.13          44         Sixta Heyden
168        41.0         0        69.38          32       Brianne Broome
169        77.0         0       101.25          47  Elouise Weatherholt
170        72.0         0        92.81          38        Addie Maharaj


In [8]:
#print a summary of the dataframe
print(data.describe())

       quiz_total    attended  final_grade  final_exam
count  171.000000  171.000000   171.000000  171.000000
mean    69.131579    0.192982    91.855380   39.672515
std      9.488473    0.395798     9.487795    5.511304
min      0.000000    0.000000     2.500000    0.000000
25%     65.500000    0.000000    89.380000   37.000000
50%     72.000000    0.000000    93.130000   40.000000
75%     75.500000    0.000000    96.360000   44.000000
max     79.500000    1.000000   104.380000   49.000000




*   The average of quiz_total means the average score of the total quizzes of the students in the dataframe.
*   The average of attended is the percentage of students who attended the class.
*   The average of final_grade is the mean point of their final grade recieved.
*   The average of fina_exam is the average of points that the students got in the final exam.





# Question 2

In [9]:
#a.Subset the data by whether someone attended class

In [10]:
data.head()

quiz_total  attended  final_grade  final_exam               name
0       78.50         1        99.38          46       Linnie Lietz
1       64.25         0        95.78          45      Mitch Mustain
2       76.75         0        97.42          41     Salina Chavera
3       69.00         1        92.50          38  Kimberely Conwell
4       64.00         0        92.50          39          Zack Burk

In [11]:
attend = data[data["attended"]==1]
not_attend = data[data['attended']==0]

In [12]:
#b.What is the average “quiz_total”, “final_grade”, “final_exam” for those who attended class versus did not attend?

print(attend.mean())

quiz_total     72.500000
attended        1.000000
final_grade    95.072121
final_exam     40.636364
dtype: float64


In [13]:
print(not_attend.mean())

quiz_total     68.326087
attended        0.000000
final_grade    91.086159
final_exam     39.442029
dtype: float64


The average quiz score of the students who sttended class is 72.5 versus about 68.33 for those who do not attend class.

The average final exam for those who attended class is 95 and the students who did not attend was 91. 

The average final exam score for those who attended class is 40 .63 and those who did not attend is 39.4.

Students who attended to class score higher in average than those who did not. 

# Bonus

In [33]:
#Find the students working together
data[data.duplicated(subset=['quiz_total','final_exam','final_grade'],keep=False)]

quiz_total  attended  final_grade  final_exam           name
146        72.0         0        92.81          38  Jaraham Eidda
170        72.0         0        92.81          38  Addie Maharaj

In [88]:
name  = data.loc[[146,170],['name']]

In [92]:
print(name)

              name
146  Jaraham Eidda
170  Addie Maharaj
